backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java

```diff
@@ -50,65 +62,127 @@ private void updatePersonFacility(Person person, UUID facilityId) {
     person.setFacility(facility);
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public List<Person> getPatients(UUID facilityId, int pageOffset, int pageSize) {
-    return _repo
-        .findByFacilityAndOrganization(
-            _os.getFacilityInCurrentOrg(facilityId),
-            _os.getCurrentOrganization(),
-            false,
-            PageRequest.of(pageOffset, pageSize, NAME_SORT))
-        .toList();
+  // Specifications filters for queries
+  private Specification<Person> inCurrentOrganizationFilter() {
+    return (root, query, cb) ->
+        cb.equal(root.get(SpecField.ORGANIZATION), _os.getCurrentOrganization());
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public long getPatientsCount(UUID facilityId) {
-    return _repo.countAllByFacilityAndOrganization(
-        _os.getFacilityInCurrentOrg(facilityId), _os.getCurrentOrganization(), false);
+  // Note: Patients with NULL facilityIds appear in ALL facilities.
+  private Specification<Person> inFacilityFilter(UUID facilityId) {
+    return (root, query, cb) ->
+        cb.and(
+            cb.equal(root.get(SpecField.ORGANIZATION), _os.getCurrentOrganization()),
+            cb.or(
+                cb.isNull(root.get(SpecField.FACILITY)), // null check first
+                cb.equal(root.get(SpecField.FACILITY).get(SpecField.INTERNAL_ID), facilityId)));
   }
 
-  @AuthorizationConfiguration.RequirePermissionSearchPatients
-  public List<Person> getAllPatients(int pageOffset, int pageSize) {
-    return _repo
-        .findAllByOrganization(
-            _os.getCurrentOrganization(), false, PageRequest.of(pageOffset, pageSize, NAME_SORT))
-        .toList();
+  private Specification<Person> nameMatchesFilter(
+      @Size(min = MINIMUM_CHAR_FOR_SEARCH) String namePrefixMatch) {
+    String likeString = namePrefixMatch.trim().toLowerCase() + "%";
+    return (root, query, cb) ->
+        cb.or(
+            cb.like(
+                cb.lower(root.get(SpecField.PERSON_NAME).get(SpecField.FIRST_NAME)), likeString),
+            cb.like(
+                cb.lower(root.get(SpecField.PERSON_NAME).get(SpecField.MIDDLE_NAME)), likeString),
+            cb.like(
+                cb.lower(root.get(SpecField.PERSON_NAME).get(SpecField.LAST_NAME)), likeString));
```

Comment:
&lt;sigh&gt; at spotless...

I would probably be tempted to factor out the boilerplate elements of these three lines into a lambda or a utility function. I would probably be making life more complicated if I did, though. 😁

Reply:
lambdas to trick spotless IS clever though.